### DISCUSSION TITLE
Creating a new repo

### DISCUSSION THREAD

**Discussion by olzama** [delph-in/docs#89] [Team Posts]
Pending the document that we plan to send to the SC, asking them to establish/clarify some policies, I will just post here with the specific action item:

I want to create a repository for my Sahaptin project that I started with Sharon (a Sahaptin grammar). Should I just create it here, within the org? (I do have the privileges at the moment, so I am not necessarily asking anyone to create it for me; just asking whether that's the right thing to do in people's opinion.)

**Comment by goodmami** [comment]
You can always move it later if we think it would be better under another owner. For this reason, I generally prefer to put things under a personal account first then move to an org if there's a reason to.

Some reasons for putting a repo under an organization:
- Publicity / Availability:
  - The project has general interest within the organization (not just a personal project)
  - It serves as a showcase of DELPH-IN technologies
- Maintainance / Collaboration:
  - The project would benefit from team permissions of people in the organization (personal repos can add individual collaborators, too, but not teams)
  - The project has no owner, the owner is not on GitHub, or the owner has moved on, and the project is still relevant

These are just my thoughts. We may make other criteria after discussing with the SC. I don't think there's any limit to the number of repos an org may have, so the org could also serve as a project ~~graveyard~~ museum.

For your Sahaptin project, I'd think you could put it under your personal account, at least to start. This would help you to get credit for it, too. For comparison, I have The [Wn](https://github.com/goodmami/wn) wordnet library under my personal account as I'm the main developer, but I plan to move it to the [globalwordnet](https://github.com/globalwordnet) or [bond-lab](https://github.com/bond-lab/) organizations if I'm unable to continue. For PyDelphin, I moved it to the delph-in org because I was hoping it might become a community-maintained project (although this hasn't worked out as well as I'd hoped).

**Comment by arademaker** [comment]
I agree with @goodmami. The Portuguese grammar we started is under our own organization that also host other Portuguese resources. Similar to the bond-lab org mentioned by @goodmami. I am planning to move the https://github.com/own-pt/wql and https://github.com/own-pt/delphin-rdf for here however, with the same hoping of @goodmami

**Comment by arademaker** [comment]
The delphin-RDF is an indirect collaboration to pydelphin, it is a plug-in for it. 

But sorry for not being a enthusiastic about Python!! ;-)

**Comment by olzama** [comment]
Hmm. Actually, if the org has no limit on the number of repos, then I don't see much benefit in **not** creating a new DELPH-IN-focused repo **not** under the org. What are the "cons"? Security, maybe? Someone malicious might delete my repo? Anything else? I would think having stuff as visible as possible would be a good thing? Suppose I start a project and then abandon it (happens...); if it is in the org, chances are higher whoever else becomes interested in the topic will notice it. Of course I can move it once I realized I am about to abandon it, but what if I forget :).  In other words, I guess I still don't see why I wouldn't create it in the org... I am not worried about not getting credit for it because it will still be listed in my Github profile, I think (not to mention it will be obvious who's been making the commits. Anything else?.. Perhaps the concern is that the org then does become a "graveyard" of tiny, unfinished things? I can see that. On the other hand, not sure whether too few projects is better than a "graveyard"/museum?

**Comment by olzama** [comment]
(P.S.: I already created the sahaptin repo using my personal account, but would still like to discuss this general issue further!)

**Comment by goodmami** [comment]
> if the org has no limit on the number of repos, 

To be clear, personal accounts don't have a limit, either.

> I don't see much benefit in **not** creating a new DELPH-IN-focused repo **not** under the org.

Ungh... too many negatives :thinking: You mean you don't see benefit in creating a DELPH-IN-focused repo in a personal account compared to under the DELPH-IN org?

Having it in your personal acct rather than the org is like saying "I own this" versus "I have worked on this". For some things, I think the former can be better. For instance, when it's in the org, it's not clear who should respond to issues or support requests.

> I would think having stuff as visible as possible would be a good thing? Suppose I start a project and then abandon it (happens...); if it is in the org, chances are higher whoever else becomes interested in the topic will notice it.

The visibility is a limited resource and with unlimited users of that resource we have a tragedy-of-the-commons situation. But we probably shouldn't think so hard about it or have strict criteria for inclusion, as that would dampen the friendly atmosphere :wink:.

We can also "pin" up to 6 repos to the front page of the organization, for those that are showcased.

**Comment by olzama** [comment]
> But we probably shouldn't think so hard about it or have strict criteria for inclusion, as that would dampen the friendly atmosphere 😉.

Agreed! :)

I like your point about ownership. I guess in some cases, this may be dictated somehow by funding or what not, or by the research group structure. In other cases, maybe the developer wants to start something within the community and cares less about ownership. This may be a guiding criterion, I suppose. E.g. for Sahaptin, my goal is to find out how to use our grammars to pursue linguistic questions; I don't care who owns the project etc., I'd rather more people see it (which may or may not happen).

> For instance, when it's in the org, it's not clear who should respond to issues or support requests.

Yes, but I think that's a problem, actually! I think it should be clear who responds :) somehow.
